### PR TITLE
Updated for Swift 2 (Xcode 7 GM)

### PR DESCRIPTION
**Pull request by buildreactive**
Updated to Swift 2 syntax. Minor docs update and pod spec. Tested on GM release of Xcode 7.

### CODE REVIEW

File: DateTimeKit.podspec [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+							DateTimeKit is a Swift library that provides simple, easy-to-use date, time and timezone manipulation.
+							The ideas behind DateTimeKit are quite heavily influenced by the JodaTime library.
+
+							DateTimeKit removes the dependence on NSDate and NSCalendar and provides a new set of objects in their
+							place.
+                   DESC
+
+  s.homepage     = "https://github.com/edwardaux/DateTimeKit"
+  s.license      = { :type => "MIT" }
+  s.authors      = { 'Daniele Margutti' => 'me@danielemargutti.com', 'Zacharias Beckman' => 'zbeckman@HyraxLLC.com' }
```
Comment (edwardaux) [inline DateTimeKit.podspec]: Maybe include me in this list too?

**Comment by edwardaux** [comment]
Can you exclude `DateTimeKit.xcodeproj/xcuserdata/zbeckman.xcuserdatad/`? We shouldn't have any user data.

File: DateTimeKit/DateTime.swift [hunk 2]
```diff
@@ -207,23 +202,89 @@ public struct DateTime {
...
+		}
+	}
+	
+	public func withZone(zone: Zone) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
```
Comment (edwardaux) [inline DateTimeKit/DateTime.swift:276]: Thanks for the additional functions. I'd prefer if you used `self.` for all of the instance variables, though, so it is much clearer which variable is shadowed.

File: DateTimeKit/DateTimeKit.h [hunk 3]
```diff
@@ -6,14 +6,14 @@
...
+//
+////! Project version number for DateTimeKit.
+//FOUNDATION_EXPORT double DateTimeKitVersionNumber;
+//
+////! Project version string for DateTimeKit.
+//FOUNDATION_EXPORT const unsigned char DateTimeKitVersionString[];
+//
+//// In this header, you should import all the public headers of your framework using statements like #import <DateTimeKit/PublicHeader.h>
+//
+//
```
Comment (edwardaux) [inline DateTimeKit/DateTimeKit.h]: I'm curious why commenting these out, rather than just deleting?

**Comment by edwardaux** [comment]
Generally speaking, though, it looks pretty good to me. Thanks for the contribution.

One of these days, I also want to get rid of the failable initialisers and use `throws` instead.

File: DateTimeKit/DateTimeKit.h (hunk 3 above)
Comment (buildreactive) [inline DateTimeKit/DateTimeKit.h]: Oops. Yes, I should have removed it – was experimenting.

File: DateTimeKit/DateTime.swift (hunk 2 above)
Comment (buildreactive) [inline DateTimeKit/DateTime.swift:276]: I'll update to use `self` and issue a new PR.

File: DateTimeKit.podspec (hunk 1 above)
Comment (buildreactive) [inline DateTimeKit.podspec]: Added:

```
'Craig Edwards' => 'edwardaux', 
```

(Did not have your email address).

**Comment by edwardaux** [comment]
Looks pretty good. Will double-check it tonight and merge.

**Comment by edwardaux** [comment]
Hmm... when I run the tests on my machine, one of them fails (see below). I've recently updated to El Capitan though, so I am trying to work out if it is:
- Has always been there
- Introduced by Swift 2
- Introduced by Xcode 7
- Introduced by El Capitan

FYI, the line that fails is:

`DTAssertEqual(LocalDate(input:"2012-12-25 12:34:56.789-01:00", format:"yyyy-MM-dd HH:mm:ss.SSSZZZ")!, year: 2012, month: 12, day: 25)`

Looks like `LocalDate` is converting that to 2012-12-26, instead of 2012-12-25. My machine is AEST (GM+10). Currently investigating...

**Comment by buildreactive** [comment]
The above test may be fragile and should be refactored:

```
DTAssertEqual(LocalDate(input:"2012-12-25 12:34:56.789-01:00", format:"yyyy-MM-dd HH:mm:ss.SSSZZZ")!, year: 2012, month: 12, day: 25)
```

No matter where you are in the world, there is a period during the day in which this test will fail.

It is the same as writing:

```
assert that the local day is equal to 12:34 in the time zone GMT-0100
```

At least, that is how I interpret it. Am I wrong? If not, then it will fail for you based on what time zone you happen to live in, and whether "the day right now, right here" is the same as "the day at 12:34 in TZ GMT-0100". (I'm making the assumption that `DTAssertEqual` is not adjusting for local time, which I believe is the case because I observed this test failing in PST if you adjust the TZ to +0600).

**Comment by edwardaux** [comment]
Sorry for the delay @zbeckman. I totally forgot about this pull request :scream:

I've just merged, and fixed the failing test case while I am at it. Thanks for the PR, and apologies again for taking so long to get around to merging.

**Pull request merged**